### PR TITLE
Rebalance eating raw meats to be less damaging to morale if you have predator mutations

### PR DESCRIPTION
**Pull request by hjk321**
#### Summary

SUMMARY: Balance "Eating raw meat does not hurt your morale as much if you have traits such as Hunter, Predator, etc ."

#### Purpose of change

So, cannibalism causes eating human-related foods to _increase_ your morale, right? Well, let's say you are a mutant with the hunter or apex predator trait. Eating raw meat still causes negative 10 fun, and some raw organs can take you as low as _-25._ Seeing how the PRED mutations are all past-threshold, it would make sense for mutants to exhibit animalistic traits of consuming raw meat, since the mutations specifically say that your mind is starting to shift to one of a beast. These mutation deserve similar balancing as cannibalism, since it makes no sense whatsoever to be penalized so heavily for playing this way if you have these mutations.

#### Describe the solution

Adds a new JSON flag called PREDATOR_FUN. When consuming an item, it checks for this flag and affects your morale based on mutations:

- Culler or Carnivore: A small +5 bonus, still likely to be negative.
- Hunter or Predator: +15, which makes raw meat net +5 morale but most raw organs still negative.
- Apex Predator: +20, enough to make some organs about neutral and normal meat significantly positive.

#### Describe alternatives you've considered

As the branch name may suggest, I initially considered making these sort of modifiers completely JSONified, instead of hardcoding them in C++. I eventually decided against it due to the high potential of regressions and technical debt.

#### Testing

I tested eating multiple raw meats with characters of different mutation levels. One thing reviewers may want to comment on is the max bonuses I set, since I didn't fully test how these modifiers stack with multiple foods.

#### Additional Context

The system checks for the flag and applies the morale boost exactly the same as with cannibalism. I do not expect any other item except the ones I have already changed in this PR to need this flag, but it can be used in mods, etc. with no ill effects.

### CODE REVIEW

File: src/character.cpp [hunk 1]
```diff
@@ -12779,16 +12779,7 @@ static const trait_id trait_SPIRITUAL( "SPIRITUAL" );
 
 bool Character::fun_to_read( const item &book ) const
 {
-    // If you don't have a problem with eating humans, To Serve Man becomes rewarding
```
Comment (hjk321) [inline src/character.cpp:12782]: Deleted this because the logic is duplicated by `book_fun_for`.

File: src/character.cpp [hunk 2]
```diff
@@ -12779,16 +12779,7 @@ static const trait_id trait_SPIRITUAL( "SPIRITUAL" );
...
-    if( ( has_trait( trait_CANNIBAL ) || has_trait( trait_PSYCHOPATH ) ||
-          has_trait( trait_SAPIOVORE ) ) &&
-        book.typeId() == itype_cookbook_human ) {
-        return true;
-    } else if( has_trait( trait_SPIRITUAL ) && book.has_flag( flag_INSPIRATIONAL ) ) {
-        return true;
-    } else {
-        return book_fun_for( book, *this ) > 0;
-    }
```
Comment (hjk321) [inline src/character.cpp:12791]: Removed this because the same logic is duplicated in the 'book_fun_for' call.

**Comment by Venera3** [comment]
Would it be possible to implement  a general "morale_mod" or similar field in mutations, which defines a list of comestible flags and the corresponding morale modifiers? That would make the system more flexible for JSON contributors and would open the door for drug intolerances and such.
In this case it would look "morale_mod":  [ "PREDATOR_FUN", 5/10/15 ].

**Comment by Zireael07** [comment]
@Venera3: Very nice proposal <3 but probably out of scope for this PR.

**Comment by hjk321** [comment]
@Venera3 I did initially plan to implement something like that (#45757) but eventually decided it wasn't a good idea. It would essentially require a string parsing system for the different types of modifiers possible in C++ (spirituality, for example, is a *3 modifier for cookbook_human). Such an overhaul should probably be done by someone more familiar with the codebase than I.

**Pull request merged**